packages/cli/src/databases/dsl/Indices.ts

```diff
@@ -48,10 +48,25 @@ export class CreateIndex extends IndexOperation {
 }
 
 export class DropIndex extends IndexOperation {
+	constructor(
+		tableName: string,
+		columnNames: string[],
+		tablePrefix: string,
+		queryRunner: QueryRunner,
+		customIndexName?: string,
+		protected skipIfMissing = false,
+	) {
+		super(tableName, columnNames, tablePrefix, queryRunner, customIndexName);
+	}
+
 	async execute(queryRunner: QueryRunner) {
-		return await queryRunner.dropIndex(
-			this.fullTableName,
-			this.customIndexName ?? this.fullIndexName,
-		);
+		return await queryRunner
+			.dropIndex(this.fullTableName, this.customIndexName ?? this.fullIndexName)
+			.catch((error) => {
+				if (error instanceof Error && error.message.includes('not found') && this.skipIfMissing) {
```

Comment:
They all throw the same error `TypeORMError: Supplied index {indexId} was not found in table {table}`. 

Re: handling differently depending on DB, I replied about this [above](https://github.com/n8n-io/n8n/pull/9817#discussion_r1673792303) - my reasoning is it's simpler to have one way of handling this. Once we remove MySQL this will no longer be an issue.

Reply:
Alright, makes sense 👍 Maybe we can be more specific and expect a `TypeORMError` instead of any generic `Error`?